core/src/main/java/io/grpc/Server.java

```diff
@@ -38,6 +39,16 @@
    */
   public abstract Server start() throws IOException;
 
+  /**
+   * Returns the address the server is listening on.  This can return null if there is no actual
+   * address or the result otherwise does not make sense.  Result is undefined after the server is
+   * terminated.
+   *
+   * @throws IllegalStateException if the server has not yet been started.
+   * @since 1.0.0
+   */
+  public abstract InetSocketAddress mainAddress();
```

Comment:
unfortunately this is not stable as well.  Anyone extending Server would see compilation errors when they update to the latest gRPC.  We typically make new methods like this non-abstract and throw `UnsupportedOperationException`